runtime/runtime.m

```diff
@@ -2118,6 +2118,16 @@ -(void) xamarinSetGCHandle: (int) gc_handle;
 	return res;
 }
 
+void
+xamarin_release_block_on_main_thread (void *obj)
+{
+	if ([NSThread isMainThread]) {
```

Comment:
I was wondering about why mixing ObjC and C/GDC but learned that `dispatch_get_main_queue ()` is iOS 8+ only :(